### PR TITLE
Add views for Models and Controllers

### PR DESCRIPTION
**Pull request by mershark**
In this task I accomplished the following:
1. Implemented the application user interface to align with the provided wireframes.
2. Employed methods from the Blog App models for efficient data processing.
3. Established a coherent navigation flow across pages in accordance with the wireframes:
      - Navigating to the URL directs users to the user list page (First wireframe).
      - Selecting a user name leads to the respective user's profile (Second wireframe).
      - Choosing "See All Posts" redirects users to a page displaying all posts by the selected user, including comments (Third wireframe).
      - Clicking on a post redirects users to the post page (Forth wireframe).
4. Emphasized the development of core app functionalities as the primary focus.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,7 @@
-<h1>Here is a list of users</h1>
\ No newline at end of file
+<% @users.each do |user| %>
+  <div>
+    <%= link_to user_path(user) do %>
+      <%= render partial: 'users/user_info', locals: { user: user } %>
```
Comment (danmainah) [inline app/views/users/index.html.erb:4]: @mershark Great job using partial to render your users, it's a perfect practice that helps the better organization of your code, for easier code reading and scaling in the future.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,22 @@
...
+
+<div>
+  <% @user.posts.each do |post| %>
+    <div>
+      <!-- Render the post_info partial -->
+      <%= render partial: 'posts/post_info',
+      locals: {post: post, user: @user, comments: post.comments} %>
+    </div>
+  <% end %>
+</div>
```
Comment (danmainah) [inline app/views/users/show.html.erb]: - [x] @mershark Great work displaying posts in the user show page, however, you are supposed to use the `most_recent_posts`  method to display the three latest posts only in this page.

File: app/views/posts/_post_info.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+</div>
+
+<% if defined?(show_comments) && show_comments && post.comments.present? %>
+<div class="comments-container">
+  <% comments.each do |comment| %>
+    <p>
+      <%= comment.user.name %>: <%= comment.text %>
+    </p>
+  <% end %>
+</div>
```
Comment (danmainah) [inline app/views/posts/_post_info.html.erb]: - [x] @mershark Great work displaying all the information on the user's post page, however, you are supposed to use the `most_recent_comments` post model method to display the five latest comments on this page.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @mershark , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implemented the views for the users and posts controller actions.
✔️  Added routes to ensure the views are displayed.
✔️ Added navigation from one page to another.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @mershark , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All views work correctly and look like given wireframe

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**